### PR TITLE
Updates to Search module

### PR DESCRIPTION
**Pull request by ben-albrecht**
Making finishing touches to Search module in preparation for 1.14 release.

Non-testing related changes are best viewed in this [commit](https://github.com/chapel-lang/chapel/commit/f03ced4b68381b9a6da82b56068be829acec778b).

### Changes include:

* **Interface changes**
  * `linearSearch()` now assumes the array is unsorted. If users are searching a pre-sorted array, they should use `binarySearch()`.
    * comparator support was left in for `linearSearch()` so that users can redefine how they want to check equality of array data without overloading `op==()` for their data type. This also keeps the `*Search()` interface consistent.
  * `search()` now supports a `sorted: bool` argument, which determines if it wraps `binarySearch()` (`sorted=true`), or `linearSearch()` (sorted=false, default value).
     * This affects all `search()` calls, since the new default behavior is to assume that the array is not pre-sorted, and thus it calls `linearSearch()` by default now instead of `binarySearch()`
  * The old `BinarySearch()` &&  `LinearSearch()` routines print a deprecation warning, and can be officially removed in 1.15.
* Documentation fixes that referred to Sort module instead of Search module
* Updated all the tests that use these interfaces.
   * Notably, removed an unused call to `linearSearch()` from `timeVectorArray.chpl`.

[*paratest sanity check completed*]

* 153 errors found - fixed [here](https://github.com/chapel-lang/chapel/pull/4273/commits/b63e309a731169e7d6d05187d19e48a00e97e670) and [here](https://github.com/chapel-lang/chapel/pull/4273/commits/fd17cb7c92cec64322d7c6ff4cc514a3e8865455)

[*second paratest  complete*]

* 0 errors

### CODE REVIEW

File: modules/packages/Search.chpl [hunk 1]
```diff
@@ -18,63 +18,65 @@
...
       been if it was not found.
    :rtype: (`bool`, `Data.domain.type`)
  */
-proc search(Data:[?Dom], val, comparator:?rec=defaultComparator) where Dom.rank == 1 {
-  return binarySearch(Data, val, comparator);
+proc search(Data:[?Dom], val, comparator:?rec=defaultComparator, sorted=false) where Dom.rank == 1 {
+  if sorted then
+    return binarySearch(Data, val, comparator);
+  else
+    return linearSearch(Data, val);
```
Comment (mppf) [inline modules/packages/Search.chpl]: shouldn't you pass comparator here, so == can be gotten from it?

**Comment by mppf** [comment]
Looks good, except for that one comment.

**Comment by e-kayrakli** [comment]
There has to be some minor performance related changes in some code I recently merged.

In `modules/dists/SparseBlockDist.chpl` method `bulkAdd_help`, there are calls to `search` where the input array is guaranteed to be sorted. It either has to be changed to `binarySearch` or `sorted=true` must be passed. Should I do that as a separate PR, or should the fix go in with this?

**Comment by ben-albrecht** [comment]
@e-kayrakli -- thanks for bringing that to my attention. I will fix that in this PR.

**Pull request merged**